### PR TITLE
Configure ESLint, fix ESLint errors

### PR DESCRIPTION
**Pull request by rhansen**
Multiple commits that set up ESLint and fix the issues it complains about:
* Configure ESLint
* Run `eslint --fix .`
* Fix comments
* Combine two string literals into one
* Delete unnecessary `else`
* Fix comment indentation
* Switch single and double quotes in HTML string
* Use `this` instead of `self` where possible
* Enable strict mode everywhere
* Delete unused imports
* Delete most unused variables
* Delete redundant variable redeclarations
* Delete commented-out code
* Delete dead code
* Use `const` and `let` instead of `var`
* Rename `ep_comments` class to `EpComments`
* Simplify handling of "Show Comments" toggle
* Use camel case for variable names
* Remove redundant ESLint global declarations
* Reduce global namespace pollution
* Use jQuery to build HTML
* Fix various minor ESLint errors:
* Don't iterate over inherited properties
* Simplify comment counting
* Rewrite `getSelectedText` for readability, robustness
* Delete redundant jQuery imports
* Avoid mutating passed-in list
* Capitalize PreCommentMarker class name
* Add missing variable declaration
* tests: Fix exception error message (copy+paste bug)
* tests: Delete unnecessary wrapper objects
* tests: Switch some function expressions to arrow functions

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -285,26 +284,26 @@ exports.expressCreateServer = function (hook_name, args, callback) {
...
-        data = JSON.parse(fields.data);
+      data = JSON.parse(fields.data);
     } catch (err) {
-      res.json({code: 1, message: "data must be a JSON", data: null});
+      res.json({code: 1, message: 'data must be a JSON', data: null});
       return;
     }
 
-    let replyIds, replies;
+    let replyIds; let replies;
```
Comment (JohnMcLear) [inline index.js:293]: surprised you prefer this syntax

File: index.js [hunk 2]
```diff
@@ -130,62 +127,64 @@ exports.socketio = function (hook_name, args, cb){
...
-  args.content = args.content + eejs.require("ep_comments_page/templates/settings.ejs");
+exports.eejsBlock_mySettings = (hookName, args, cb) => {
+  args.content += eejs.require('ep_comments_page/templates/settings.ejs');
   return cb();
 };
 
-exports.padInitToolbar = function (hookName, args) {
-  var toolbar = args.toolbar;
+exports.padInitToolbar = (hookName, args) => {
+  const toolbar = args.toolbar;
```
Comment (JohnMcLear) [inline index.js:141]: just a check to ensure toolbar isn't mutated when ``registerButton`` is fired.

File: static/js/commentBoxes.js [hunk 3]
```diff
@@ -1,125 +1,126 @@
...
     if (targetLeft + modalWitdh > containerWidth) {
       targetLeft = containerWidth - modalWitdh - 25;
     }
-    var editorCommentHeight = editorComment ? editorComment.outerHeight(true) : 30;
+    const editorCommentHeight = editorComment ? editorComment.outerHeight(true) : 30;
     getPadOuter().find('.comment-modal').addClass('popup-show').css({
-      left: targetLeft + "px",
-      top: targetTop + editorCommentHeight +"px"
+      left: `${targetLeft}px`,
+      top: `${targetTop + editorCommentHeight}px`,
```
Comment (JohnMcLear) [inline static/js/commentBoxes.js:94]: til you can do logic in templates

File: static/js/commentIcons.js [hunk 4]
```diff
@@ -1,222 +1,216 @@
...
-var displayIcons = function() {
-  return clientVars.displayCommentAsIcon
-}
+const displayIcons = () => clientVars.displayCommentAsIcon;
 
 // Easier access to outer pad
-var padOuter;
-var getPadOuter = function() {
+let padOuter;
+const getPadOuter = () => {
```
Comment (JohnMcLear) [inline static/js/commentIcons.js:10]: In general I quite like ``function(){`` so can we not be overly strict about this?  I prefer the syntax and I think it's more descriptive, just my personal opinion.  Can merge w/ your changes in, I don't hate them, I just don't mind either approach 👍

**Comment by JohnMcLear** [comment]
So a few things as we discussed..

1) awesome work!
2) I'd really like in the future to see fixes land separate from test modifications, just so we can see that the fixes/patches land and that the software functions as it was before.
3) I'm not strict about ``function(){}`` vs ``() =>`` and for me that's fine.

TODO: Tests are failing :)  Will need fixing up before merging!

File: index.js (hunk 1 above)
Comment (rhansen) [reply index.js]: I don't; thanks for pointing that out. Fixed.

File: index.js (hunk 2 above)
Comment (rhansen) [reply index.js:141]: `const` only prevents reassignment of that variable to another value—it doesn't mean that the object is immutable.

File: static/js/commentIcons.js (hunk 4 above)
Comment (rhansen) [reply static/js/commentIcons.js:10]: `function` and arrow functions [do not behave the same](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions), so it's not just a matter of visual preference.

Good style is to only use `function` for class methods (though see [ES6 method shorthand](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions)) and arrow functions for everything else. There are exceptions, such as callbacks passed to [jQuery's `.each()` method](https://api.jquery.com/each/) and certain event handlers.

**Comment by rhansen** [comment]
> TODO: Tests are failing

Yup, though they are failing the same way on `master`.

**Comment by JohnMcLear** [comment]
Nice, afaik @ilmartyrk is working on the failing tests.

**Pull request merged**